src/code/z_player_lib.c

```diff
@@ -1,153 +1,1046 @@
+        }
+    }
+
+    return 0;
+}
+
+s32 func_80124148(Player* player) {
+    return func_80124110(player, player->itemActionParam);
+}
+
+s32 Player_ActionToMeleeWeapon(s32 actionParam) {
+    s32 weapon = actionParam - PLAYER_AP_UNK_2;
+
+    if ((weapon > (PLAYER_AP_UNK_2 - PLAYER_AP_UNK_2)) && (weapon <= (PLAYER_AP_UNK_8 - PLAYER_AP_UNK_2))) {
+        return weapon;
+    }
+    return 0;
+}
+
+s32 Player_GetMeleeWeaponHeld(Player* player) {
+    return Player_ActionToMeleeWeapon(player->itemActionParam);
+}
+
+s32 Player_HoldsTwoHandedWeapon(Player* player) {
+    // Relies on two-handed weapons to be contiguous
+    if ((player->itemActionParam >= PLAYER_AP_SWORD_GREAT_FAIRY) && (player->itemActionParam <= PLAYER_AP_STICK)) {
+        return true;
+    }
+
+    return false;
+}
+
+s32 Player_ActionToBottle(Player* player, s32 actionParam) {
+    s32 bottle = actionParam - PLAYER_AP_BOTTLE;
+
+    // Relies on bottle-related action params to be contiguous
+    if ((bottle >= (PLAYER_AP_BOTTLE - PLAYER_AP_BOTTLE)) && (bottle <= (PLAYER_AP_BOTTLE_FAIRY - PLAYER_AP_BOTTLE))) {
+        return bottle;
+    }
+
+    return -1;
+}
+
+s32 Player_GetBottleHeld(Player* Player) {
+    return Player_ActionToBottle(Player, Player->itemActionParam);
+}
+
+s32 Player_ActionToExplosive(Player* player, s32 actionParam) {
+    s32 explosive = actionParam - PLAYER_AP_BOMB;
+
+    // Relies on explosive-related action params to be contiguous
+    if ((explosive >= (PLAYER_AP_BOMB - PLAYER_AP_BOMB)) && (explosive <= (PLAYER_AP_BOMBCHU - PLAYER_AP_BOMB))) {
+        return explosive;
+    }
+
+    return -1;
+}
+
+s32 Player_GetExplosiveHeld(Player* player) {
+    return Player_ActionToExplosive(player, player->itemActionParam);
+}
+
+s32 func_80124278(Actor* actor, s32 arg1) {
+    s32 phi_v1 = 0;
+
+    // Fake match?
+    if ((arg1 == 1) || ((phi_v1 = arg1 - 3, (phi_v1 >= 0)) && (phi_v1 < 4))) {
+        return phi_v1;
+    }
+
+    return -1;
+}
+
+s32 func_801242B4(Player* player) {
+    return (player->stateFlags1 & PLAYER_STATE1_8000000) && player->currentBoots < PLAYER_BOOTS_ZORA_UNDERWATER;
+}
+
+s32 func_801242DC(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    TextTriggerEntry* triggerEntry;
+    s32 index;
+
+    if (globalCtx->roomCtx.currRoom.unk2 == 3) { // Room is hot
+        index = 0;
+    } else if ((player->transformation != PLAYER_FORM_ZORA) && (player->unk_AD8 > 80)) {
+        index = 3;
+    } else if (player->stateFlags1 & PLAYER_STATE1_8000000) {
+        if ((player->transformation == PLAYER_FORM_ZORA) && (player->currentBoots >= PLAYER_BOOTS_ZORA_UNDERWATER) &&
+            (player->actor.bgCheckFlags & 1)) {
+            index = 1;
+        } else {
+            index = 2;
+        }
+    } else {
+        return 0;
+    }
+
+    // Trigger general textboxes under certain conditions, like "It's so hot in here!". Unused in MM
+    triggerEntry = &sEnvironmentTextTriggers[index];
```

Comment:
```suggestion
    triggerEntry = &sEnvironmentTextTriggers[envIndex];
```
Maybe to give it a more interesting name?